app/controllers/application_controller.rb

```diff
@@ -323,4 +324,8 @@ def handle_unverified_request
     logger.info("unverified_request session:" + session.inspect)
     reset_session
   end
+
+  def store_auto_publish_url
+    Thread.current[:auto_publish_url] = "#{request.protocol}#{request.host_with_port}"
```

Comment:
@scytacki should we use a protocol relative URL here?  This is called the `auto_publish_url`, but its actually the authoring server, eventually It's used in `lightweight_activity.rb`: 

     def serialize_for_portal(host)
        local_url = "#{host}#{Rails.application.routes.url_helpers.activity_path(self)}"
        …
     end